src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -38,9 +38,10 @@ internal class SubstitutedMethodSymbol : WrappedMethodSymbol
 
         private int _hashCode; // computed on demand
 
-        internal SubstitutedMethodSymbol(SubstitutedNamedTypeSymbol containingSymbol, MethodSymbol originalDefinition)
+        internal SubstitutedMethodSymbol(NamedTypeSymbol containingSymbol, MethodSymbol originalDefinition)
             : this(containingSymbol, containingSymbol.TypeSubstitution, originalDefinition, constructedFrom: null)
         {
+            Debug.Assert(containingSymbol is SubstitutedNamedTypeSymbol || containingSymbol is SubstitutedErrorTypeSymbol);
```

Comment:
Error symbols are designed specifically to be "used as if they are good" to avoid having to special-case error recovery code. As such, I believe this is precisely the correct fix (as opposed to adding special-case error recovery code in each caller).